apps/language_server/test/experimental/provider/handlers/goto_implementation_test.exs

```diff
@@ -0,0 +1,63 @@
+defmodule ElixirLS.LanguageServer.Experimental.Provider.Handlers.GotoImplementationTest do
+  use ExUnit.Case, async: false
+
+  alias ElixirLS.LanguageServer.Experimental.Protocol.Requests.GotoImplementation
+  alias ElixirLS.LanguageServer.Experimental.Protocol.Types, as: LSPTypes
+  alias ElixirLS.LanguageServer.Experimental.Provider.Env
+  alias ElixirLS.LanguageServer.Experimental.Provider.Handlers
+  alias ElixirLS.LanguageServer.Experimental.SourceFile
+  alias ElixirLS.LanguageServer.Experimental.SourceFile.Conversions
+
+  alias ElixirLS.LanguageServer.Fixtures.LspProtocol
+  alias ElixirLS.LanguageServer.Test.FixtureHelpers
```

Comment:
Hello, Steve, It is already sorted by `mix recode`; the result shows only `find_references_test.exs` needs to be sorted.

<img width="910" alt="image" src="https://user-images.githubusercontent.com/12830256/223892909-14d095b1-3877-4a75-9e0a-b20b113b1508.png">

And I don't understand why we need to put the `use ExUnit.Case, async: false` after the aliases, I saw in the `find_references_test.exs` file you didn't do that too.

Can we leave it? Because I think put the `use ExUnit.Case, async: false` to the top, We can see at a glance that this is a test file, and our eyes can quickly recognize it when opening multiple windows, and this is also more friendly for templates, by which I mean templates for quick test creation. something like: https://thoughtbot.com/blog/supercharge-elixir-and-phoenix-navigation-with-vim-projectionist

Reply:
It matters where `use` goes, and if you put use at the top of a file, then that becomes a convention, and then you either can't alias a module you `use` or all your `use`s won't be together. I vastly prefer all `use`s (and `alias`es, and `require`s, and `import`s) to be together, and for them to have short names. This project already has precedent for that, so let's not mess with precedent. This is also an appeal to authority, but I did ask José what the order of alias / use / require / import should be and the answer was alias first, then require, then import, then use. 

Here's what I mean, which do you prefer? Say we need to `use` `Protocol` and also `alias` it
```elixir
defmodule MyTestCase do 
  use ExUnit.Case, async: false 
  alias ElixirLS.LanguageServer.Experimental.Protocol
  use Protocol
 ...
 end
 ```
or 
```elixir 
defmodule MyTestCase do 
  use ElixirLS.LanguageServer.Protocol
  use ExUnit.Case

  alias ElixirLS.LanguageServer.Protocol
end
```
or
```elixir
defmodule MyTestCase do 
  alias ElixirLS.LanguageServer.Protocol

  use ExUnit.Case, async: false
  use Protocol
 ...
 end
```

I prefer the last one.

We already know that this is a test case, because it's in the `test` directory, and both the module name and the file name end with `test`. Using the logic above, since you're looking at the first line anyways, you'll see the name of the module. 

Furthermore, tests look different than elixir files, so I'm not too worried about being confused.

> I saw in the find_references_test.exs file you didn't do that too.

I'm not perfect, and I'll change that use then.

> and this is also more friendly for templates, by which I mean templates for quick test creation. something like:
It's really not more friendly at all, a template can still be the following:
```json
   "template": [
     "defmodule {camelcase|capitalize|dot}Test do",
     "  alias {camelcase|capitalize|dot}",
     "",
     "  use ExUnit.Case, async: true",
     "",
     "end"
   ]
```
And it's what I'm looking for.

Reply:
Thanks for the detailed explanation, `use` is indeed more flexible and easier to read when put at the end.

Also, thanks for reminding me about the template, I have now changed my template to

```json
{
  "lib/language_server/*.ex": {
    "type": "src",
    "alternate": "test/{}_test.exs",
    "template": [
      "defmodule ElixirLS.LanguageServer.{camelcase|capitalize|dot} do",
      "end"
    ]
  },
  "test/*_test.exs": {
    "type": "test",
    "alternate": "lib/language_server/{}.ex",
    "template": [
      "defmodule ElixirLS.{camelcase|capitalize|dot}Test do",
      "  alias ElixirLS.LanguageServer.{camelcase|capitalize|dot}",
      "",
      "  use ExUnit.Case, async: true",
      "end"
    ]
  }
}

```